slack_adapter/slack_helper.py

```diff
@@ -21,4 +24,35 @@ def activity_to_slack(activity: Activity) -> NewSlackMessage:
                 if att.name == 'blocks':
                     message.blocks = [att.content]
                 else:
-                    new_attachment = Object(author_name= att.name, thumb_url=att.thumbnail_url)
\ No newline at end of file
+                    new_attachment = Object(author_name= att.name, thumb_url=att.thumbnail_url)
+
+    @staticmethod
+    def command_to_activity_async(slack_body: SlackRequestBody, client: SlackClientWrapper, cancellation_token):
+        """ Creates an activity based on a slack event related to a slash command
+        :param slack_body: The data of the slack event.
+        :param client: The Slack client.
+        :param cancellation_token: A cancellation token for the task.
+        :return: An activity containing the event data.
+        """
+        if not slack_body:
+            raise Exception("slack_body")
+
+        new_conversation_account = ConversationAccount(id=slack_body.channel_id)
+        new_channel_account_from = ChannelAccount(user_id=slack_body.user_id)
+        new_channel_account_recipient = ChannelAccount(id=None)
+
+        # create activity
+        activity = Activity(id=slack_body.trigger_id,
+                            timestamp=datetime.utcnow(),
+                            channel_id="slack",
+                            conversation=new_conversation_account,
+                            from_property=new_channel_account_from,
+                            recipient=new_channel_account_recipient,
+                            channel_data=slack_body,
+                            text=slack_body.text,
+                            type=ActivityTypes.event)
+
+        activity.recipient.id = client.get_bot_user_by_team(activity, cancellation_token)

```

Comment:
Can you check if we will need to implement an await here? As the C# version does. 

FYI: https://docs.python.org/3/library/asyncio-task.html#awaitables

Reply:
You're right. The `await `keyword was needed; also, added `async` to the method declaration. Thanks!